File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Diagrams/getconnectorsinfo.rb
module Asposediagramjava
  module GetConnectorsInfo
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        connectors = diagram.getPages().getPage(0).getConnects()
        
        i = 0
        while i < connectors.getCount()
            connector = connectors.get(i)
            # Display information about the Connectors
            puts "From Shape ID : " + connector.getFromSheet().to_s
            puts "To Shape ID : " + connector.getToSheet().to_s

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithDiagrams/GetConnectorsInfo.php
<?php
namespace Aspose\Diagram\WorkingwithDiagrams;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class GetConnectorsInfo{

    public static function run($dataDir=null){

        # Crezate instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        $connectors = $diagram->getPages()->getPage(0)->getConnects();

        $i = 0;
        while ($i<sizeof($connectors->getCount())) {
        $connector =$connectors->get($i);
        # Display information about the Connectors
        print "From Shape ID : ".(string)$connector->getFromSheet().PHP_EOL;
        print "To Shape ID : ".(string)$connector->getToSheet().PHP_EOL;
        $i+=1;
        }
    }
}